tracker/progress.go

```diff
@@ -35,8 +35,22 @@ type Progress struct {
 	// entries with indices in (Match, Next) interval are already in flight.
 	//
 	// Invariant: 0 <= Match < Next.
+	// NB: it follows that Next >= 1.
 	Next uint64
 
+	// pendingCommit is the highest commit index <= Next-1 in flight to the
+	// follower.
+	//
+	// The actual in-flight commit index can be higher, but we track only up to
+	// Next-1, because higher indices can be ignored by the peer or reduced to
+	// Next-1 when all the in-flight entries are appended. When Next regresses,
+	// pendingCommit regresses too.
+	//
+	// In StateSnapshot, pendingCommit == PendingSnapshot == Next-1.
+	//
+	// Invariant: 0 <= pendingCommit < Next.
+	pendingCommit uint64
```

Comment:
Should this be exported? The `Progress` struct is part of the public API via `Status()`. Although I suppose we already have private members via `Inflights`.

Reply:
I'm biased towards unexporting fields by default, to avoid the risk of introducing unintended API expansion.

Reply:
More generally, I think `Progress` needs to be split into internal and external API. For instance, all the state modifying methods are "internal" and don't need to be exported, as well as a few fields here.

Reply:
Yes, but mixing private fields makes the struct a bit less convenient for external callers, e.g. it's not possible to do struct equality comparisons -- that's already not possible here though.

Being more deliberate about internal/external API here makes sense, e.g. by using a separate `ProgressStatus` struct for the public data returned by `Status()`.

Reply:
Filed #175 to this extent.